src/main/host/descriptor/descriptor.c

```diff
@@ -70,49 +70,72 @@ void descriptor_ref(gpointer data) {
     MAGIC_ASSERT(descriptor);
 
     (descriptor->refCountStrong)++;
```

Comment:
Is it an error to increment from 0, even if a non-zero weak count has kept the object alive? If so, maybe add a utility_assert?

Reply:
I think it would be best as an error. I added a `utility_assert()` here.